python/ray/rllib/utils/filter.py

```diff
@@ -189,49 +222,46 @@ def __repr__(self):
             self.clip, self.rs, self.buffer)
 
 
+class ConcurrentMeanStdFilter(MeanStdFilter):
+    is_concurrent = True
+
+    def __init__(self, *args, **kwargs):
+        super(ConcurrentMeanStdFilter, self).__init__(*args, **kwargs)
+        self._lock = threading.RLock()
+
+        def lock_wrap(func):
+            def wrapper(*args, **kwargs):
+                with self._lock:
+                    return func(*args, **kwargs)
+            return wrapper
+
+        self.__getattribute__ = lock_wrap(self.__getattribute__)
+
+    def lockless(self):
```

Comment:
Usually lockless means still thread-safe but even more optimized.

Is this just for serialization? Maybe add a `as_serializable` method all filters should implement instead.